pytorch_pfn_extras/profiler/time_summary.py

```diff
@@ -0,0 +1,138 @@
+import atexit
+from contextlib import contextmanager
+import time
+from typing import Tuple
+from threading import Lock, Thread
+from queue import Queue
+import multiprocessing as mp
+import torch
+from pytorch_pfn_extras.reporting import DictSummary
+
+
+Events = Tuple[torch.cuda.Event, torch.cuda.Event]
+
+
+class _CPUWorker(object):
+    def __init__(self, add, max_queue_size: int = 1000):
+        self._add = add
+        self._queue = mp.JoinableQueue(max_queue_size)
+        self._thread = Thread(target=self._worker)
+        self._thread.setDaemon(True)
+        self._thread.start()
+
+    def close(self):
+        self.wait()
+        self._queue.put(None)
+        self._thread.join()
+        self._queue.close()
+        self._queue.join_thread()
+
+    def wait(self):
+        self._queue.join()
+
+    def _worker(self):
+        while True:
+            v = self._queue.get()
+            if v is None:
+                self._queue.task_done()
+                break
+            name, value = v
+            self._add(name, value)
+            self._queue.task_done()
```

Comment:
How about using `empty()` to check if all tasks are done instead of putting `None` as a sentinel for the task queue?

```suggestion
        while not self._queue.empty():
            name, value = self._queue.get()
            self._add(name, value)
            self._queue.task_done()
```

Reply:
unfortunately this will not execute the loop at all, this runs in a thread and it needs to loop always until the centinel is retrieved :(

So when the thread begins, the queue is empty and it never consumes the value.